internal/handlers/create_catalog.go

```diff
@@ -259,6 +264,28 @@ func (h *CreateCatalogHandler) transportFromRegistry(registry *v1alpha1.Registry
 	return transport
 }
 
+// deleteObsoleteImages deletes images that are not present in the discovered registry anymore.
+func (h *CreateCatalogHandler) deleteObsoleteImages(ctx context.Context, existingImageNames sets.Set[string], newImagesNames sets.Set[string], namespace string) error {
+	for existingImageName := range existingImageNames {
+		if newImagesNames.Has(existingImageName) {
+			continue
+		}
```

Comment:
nit: isn't there an "intersection"-like method provided by this `Set` struct?

Reply:
We could use `sets.Difference` but it doesn't change much as we need to cycle to delete the images anyway. Also, this way we will cycle twice.